go/libkb/pseudonym.go

```diff
@@ -30,17 +29,26 @@ type TlfPseudonymInfo struct {
 	ID      [16]byte
 	KeyGen  int
 	HmacKey [32]byte
+
+	// Ignored in requests. Set in server responses. Using untrusted data
+	// during decryption is safe because we don't rely on TLF keys for sender
+	// authenticity. (Note however that senders must not use untrusted keys, or
+	// else they'd lose all privacy.)
+	UntrustedCurrentName string
 }
 
 func (t TlfPseudonym) String() string {
 	return hex.EncodeToString(t[:])
 }
 
 func (t *TlfPseudonym) Eq(r *TlfPseudonym) bool {
-	if t != nil && r != nil {
-		return bytes.Equal(t[:], r[:])
+	if t == nil || r == nil {
```

Comment:
Why not make this a function between value (i.e. not pointer) types? Can't think offhand any place where we need pointers-to-TlfPseudonym.